mfai/torch/models/__init__.py

```diff
@@ -9,6 +9,7 @@
 from .swinunetr import SwinUNETR
 from .unet import CustomUnet, UNet
 from .unetrpp import UNETRPP
+from .nlam import GraphLAM, HiLAM
 
 all_nn_architectures = (
```

Comment:
could we automate the definition of `all_nn_architectures` ?

Reply:
I tried using importlib but it is a bit complex and we have for instance subclasses of ModelABC in the nlam/models.py which have themselves sulbclasses so an automated import machinery would need to not import them. Honestly the added complexity outweights the benefits.

It looked like this (the fix for graph intermediate class is NOT INCLUDED):

```python
from .base import ModelABC


registry = dict()
package = importlib.import_module("mfai.torch.models")
for _, name, _ in pkgutil.walk_packages(package.__path__, package.__name__ + "."):
    module = importlib.import_module(name)
    for object_name, kls in module.__dict__.items():
        if isinstance(kls, type) and issubclass(kls, ModelABC) and kls != ModelABC:
            if kls.__name__.lower() in registry:
                raise ValueError(
                    f"Model {kls.__name__} from plugin {object_name} already exists in the registry."
                )
            registry[kls.__name__.lower()] = kls
all_nn_architectures = list(registry.values())
```

your thoughts ?

Reply:
I think this particular piece of code is rather nice, and maybe not _that_ complex, once isolated (e.g in a function ?) and commented ? Maybe if we wish to get only concrete classes ("leaves" in the class tree) then we can use the `__subclasses__()` attribute and only select a `kls` if its list of subclasses is empty ? This might be just another filter to the main `if` statement. I may miss some subtle things there, but I am optimistic. The gains would actually be strong, as adding a model would actually only need to define it (and not to add it to a registry, which can be manually error-prone).

Reply:
```
from .base import ModelABC


registry = dict()
package = importlib.import_module("mfai.torch.models")
for _, name, _ in pkgutil.walk_packages(package.__path__, package.__name__ + "."):
    module = importlib.import_module(name)
    for object_name, kls in module.__dict__.items():
        # iterating in the module and looking at all concrete classes of ModelABC
        if isinstance(kls, type) and issubclass(kls, ModelABC) and len(kls.__subclasses__())==0:
            if kls.__name__.lower() in registry:
                raise ValueError(
                    f"Model {kls.__name__} from plugin {object_name} already exists in the registry."
                )
            registry[kls.__name__.lower()] = kls
all_nn_architectures = list(registry.values())
```

Reply:
I also like the fact that if a model does not inherit from ModelABC, it won't be loaded into the registry (we can even raise an Error explaining how the registry is built).